apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/file/CustomScalarBuilder.kt

```diff
@@ -16,7 +16,7 @@ class CustomScalarBuilder(
 ): JavaClassBuilder {
   private val layout = context.layout
   private val packageName = layout.typePackageName()
-  private val simpleName = prefixBuiltinScalarNames(layout.compiledTypeName(name = customScalar.name))
+  private val simpleName = prefixBuiltinScalarNames(layout.compiledTypeName(name = customScalar.targetName))
```

Comment:
This is not needed because `compiledTypeName` uses `nameToClassName` under the hood

Reply:
More generally, I'd like `CodegenLayout` to take GraphQL names as input and outputs Kotlin names

Reply:
👍 I've reverted to using `name` in builders, and map them to `targetName` via `nameToClassName` in c0d9274f1b10a7244b88e31d50a8d98e6d6ac2d6